src/utils.ts

```diff
@@ -99,17 +99,26 @@ export type BulkWriteOperation<TSchema, TOptions extends SchemaOptions<TSchema>>
       deleteMany: DeleteManyModel<TSchema>;
     };
 
+type FilterKeys<T, U> = { [P in keyof T]: T[P] extends U ? P : never }[keyof T];
+type FilterProperties<T, U> = { [K in FilterKeys<T, U>]: T[K] };
```

Comment:
Please use longer names for the generic types here. And add some newlines between each type.

Reply:
Done!